src/base_controller.coffee

```diff
@@ -0,0 +1,23 @@
+
+# Offers a base controller to extend
+class BaseController
+
+  #helper for use in subclass constructors
+  useMiddleware: (middleware, options = {}) ->
+    {only, except} = options
+    actions = [].concat only if only?
+    actions ?= @_actions()
+    except ?= []
+    except = [].concat except
+    for action in actions when except.indexOf(action) < 0
+      @middleware ?= {}
+      @middleware[action] ?= []
+      @middleware[action] = [].concat @middleware[action]
+      @middleware[action].push middleware
+
+
+  _actions: ->
+    (action for action, _ of @ when action isnt 'useMiddleware' and action.indexOf('_') isnt 0)
```

Comment:
I think this is actually a useful method for the base controller to have. What do you think about exposing it as a public method called `getActions` and handle the `only`/`except` filtering here?


What do you think about something like this:
```coffeescript
  getActions: ({only, except} = {}) ->
    only = [].concat only ? []
    except = [].concat except ? []

    for action of this
      continue if BaseController::[action]?
      continue unless action in only or only.length is 0
      continue if action in except
      action
```